### PR TITLE
Runtime compressed refs work

### PR DESCRIPTION
**Pull request by gacholio**
Update SlotObject for runtime compressed refs.

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
@dmitripivkine @amicic

**Comment by rwy0717** [comment]
@genie-omr build all

File: gc/base/SlotObject.hpp [hunk 1]
```diff
@@ -48,24 +51,44 @@ class GC_SlotObject
...
 	convertPointerFromToken(fomrobject_t token)
 	{
+		uintptr_t value = (uintptr_t)token;
 #if defined (OMR_GC_COMPRESSED_POINTERS)
-		return (omrobjectptr_t)((uintptr_t)token << _compressedPointersShift);
-#else
-		return (omrobjectptr_t)token;
-#endif
+		if (compressObjectReferences()) {
+			value <<= _compressedPointersShift;
```
Comment (amicic) [inline gc/base/SlotObject.hpp]: did you consider always shifting, but with 0 for nonCR?

Comment (gacholio) [reply gc/base/SlotObject.hpp]: The problem is that the _compressedPointerShift field only exists if the compressed ifdef is true. In the future, if we stop ifdeffing the fields, I think the code would look better as you suggest.

**Pull request merged**